### PR TITLE
fix: graceful shutdown when INX connection breaks

### PR DESCRIPTION
**Pull request by jorgemmsilva**

### CODE REVIEW

File: tools/cluster/tests/env.go [hunk 1]
```diff
@@ -18,13 +18,15 @@ import (
 
 // TODO remove this?
 func setupWithNoChain(t *testing.T, opt ...waspClusterOpts) *ChainEnv {
-	return &ChainEnv{t: t, Clu: newCluster(t, opt...)}
+	clu, _ := newCluster(t, opt...)
+	return &ChainEnv{t: t, Clu: clu}
```
Comment (muXxer) [inline tools/cluster/tests/env.go:22]: any reason why you didn't pass the dataPath here, but did in another changed code part?

File: clients/multiclient/node.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package multiclient
+
+import (
+	"context"
+
+	"github.com/iotaledger/wasp/clients/apiclient"
+)
+
+// ActivateChain sends a request to activate a chain in all wasp nodes
```
Comment (muXxer) [inline clients/multiclient/node.go]: Copy paste error :trollface:

File: packages/nodeconn/nodeconn.go [hunk 3]
```diff
@@ -707,7 +726,8 @@ func (nc *nodeConnection) AttachChain(
 	}()
 
 	if err := chain.SyncChainStateWithL1(ctx); err != nil {
-		nc.LogPanicf("synchronizing chain state %s failed: %s", chainID, err.Error())
+		nc.LogError("synchronizing chain state %s failed: %s", chainID, err.Error())
+		nc.shutdownHandler.SelfShutdown(fmt.Sprintf("Cannot sync chain %s with L1", chain.chainID), true)
```
Comment (muXxer) [inline packages/nodeconn/nodeconn.go]: Maybe also add the error message here?

File: tools/cluster/tests/env.go (hunk 1 above)
Comment (jorgemmsilva) [reply tools/cluster/tests/env.go:22]: so it is available in `chainEnv`.
a lot of tests just start with `setupNativeInccounterTest` that return a `chainEnv`, this makes the chainEnv aware of the data path

**Review by muXxer — APPROVED** [review]

**Pull request merged**